### DISCUSSION TITLE
Help with a simulating Drive Cycle from a BPX Schema

### DISCUSSION THREAD

**Discussion by MarkVeerasingam** [pybamm-team/PyBaMM#3698] [Q&A]
Hi, all I'm trying to run this code from a BPX model here: [About: Energy BPX Parameterization ](https://github.com/About-Energy-OpenSource/About-Energy-BPX-Parameterisation/blob/main/NMC/validation-plots.ipynb)

However I know since timescales been removed, however this didn't work for me

```
# Read drive cycle data from a CSV file
drive_cycle = pd.read_csv(r"NMC_validation\NMC_25degC_DriveCycle.csv", comment="#", header=None).to_numpy()

# split out time [s] vs voltage [V]
voltage_data = drive_cycle[:, [0, 2]]

# split out time [s] vs current [A]
current_data = drive_cycle[:, [0, 1]]

current_interpolant = pybamm.Interpolant(
        current_data[:, 0], -current_data[:, 1], pybamm.t, interpolator="linear")
```
        
It's returning this error:
---> [11](vscode-notebook-cell:?execution_count=47&line=11)         current_data[:, 0], -current_data[:, 1], pybamm.t, interpolator="linear")

TypeError: bad operand type for unary -: 'str'

And I have no idea on how about resolving this, if anyone could help me out that would be greatly appreciated. I'm currently trying to learn more other parts of pybamm like BPX and other modelling tools and really optimistic for BPX.
 You can find the drive cycle data [here](https://github.com/About-Energy-OpenSource/About-Energy-BPX-Parameterisation/blob/main/NMC/data/validation/NMC_25degC_DriveCycle.csv)

**Comment by TomTranter** [comment]
You should try with the new experiment step class https://docs.pybamm.org/en/stable/source/api/experiment/experiment_steps.html which accepts arrays as values

**Reply by TomTranter** [reply]
https://github.com/pybamm-team/PyBaMM/blob/develop/docs/source/examples/notebooks/getting_started/tutorial-5-run-experiments.ipynb
